centreon/centreon-stream-connector-scripts
# fix splunk max buffer size

## Description

there is a huge bug with the max_buffer size and the splunk metrics stream connector. 

What we want: 

- if max_buffer_size is set to 2, we only want to send 2 metrics in a payload.

What we have:

- if a service has a thousand of metrics, the payload will contain every single metric of said service


## Type of change


## Target serie

- [x] 23.04.x

<h2> How this pull request can be tested ? </h2>

- configure this stream connector with a max_buffer_size of 3 for example
- use the send_data_test parameter (set it to 1)
- make sure that a payload only has at best three metrics
- make sure that a payload only has metrics from a given service
- make sure that you receive all the data that you are supposed to receive (for example, you schedule a check on two different ping service, you must receive the data from both ping services (make sure they are on different hosts or they have a different name in order to be able to differentiate them)
